components/ledger/internal/storage/ledgerstore/balances.go

```diff
@@ -27,6 +28,7 @@ func (store *Store) GetAggregatedBalances(ctx context.Context, q *GetAggregatedB
 				NewSelect().
 				Table(MovesTableName).
 				ColumnExpr("distinct on (moves.account_address, moves.asset) moves.*").
+				Join("left join accounts_metadata am on am.address = moves.account_address").
```

Comment:
Maybe it's not possible in the code, but can we join only if we aggregate using metadata instead of all queries ?

Reply:
It should be possible.